reya-rust-sdk/src/http_provider.rs

```diff
@@ -738,15 +745,15 @@ fn decode_auto_exchange_error(reason_bytes: Bytes) -> (String, AEReasonError) {
     }
 }
 
-fn handle_auto_exchange_error(e: Error) {
+fn handle_rpc_error(e: Error) -> Option<String> {
     match e {
         Error::AbiError(revert_reason) => {
-            info!("[Error decoding] ABI error: {:?}", revert_reason);
+            error!("[Error decoding] ABI error: {:?}", revert_reason);
         }
         Error::TransportError(error) => {
             let RpcError::ErrorResp(e) = error else {
-                info!("[Error decoding] Failed to match ErrorResp {:?}", error);
-                return;
+                error!("[Error decoding] Failed to match ErrorResp {:?}", error);
+                return None;
```

Comment:
why not return the same error string as in the error! and on the other places too?

Reply:
I want to obtain the decoded contract error, if I fail to get it, that will just be an unknown error. Other logs will tell you why it failed to decode